app/src/main/kotlin/at/bitfire/davdroid/resource/LocalAddressBookStore.kt

```diff
@@ -47,21 +47,21 @@ class LocalAddressBookStore @Inject constructor(
      *
      * The address book account name contains
      *
-     * - the collection display name or last URL path segment
+     * - the collection display name or last URL path segment (filtered for special characters)
      * - the actual account name
      * - the collection ID, to make it unique.
      *
      * @param info  Collection to take info from
      */
     fun accountName(info: Collection): String {
-        // Name the address book after given collection display name, otherwise use last URL path segment
-        val sb = StringBuilder(info.displayName.let {
-            if (it.isNullOrEmpty())
-                info.url.lastSegment
-            else
-                it
-        })
+        // Name of address book is given collection display name, otherwise the last URL path segment
+        var name = info.displayName.takeIf { !it.isNullOrEmpty() } ?: info.url.lastSegment
+
+        // Remove every character that is not alphanumeric or "-", "_", " "
+        name = name.replace(Regex("[^a-zA-Z0-9-_ ]"), "")
```

Comment:
Guava provides CharMatcher, which may be suitable for our case. I wonder whether we really should exclude everything except Latin characters/numbers. What about for instance Cyrillic or Chinese characters? If they're generally supported by Android, I'd not remove them without reason.

Maybe we can [remove](https://guava.dev/releases/snapshot/api/docs/com/google/common/base/CharMatcher.html#removeFrom-java.lang.CharSequence-) ISO [control characters](https://guava.dev/releases/snapshot/api/docs/com/google/common/base/CharMatcher.html#javaIsoControl()) + a few that could cause SQL problems (backtick, `'`, `"`)?

Reply:
Yes the problem is very rare, and until now was always related to apostrophes. So I hope that with filtering a few characters the most occurrences will vanish.